src/qibo/models/dbi/double_bracket.py

```diff
@@ -114,47 +146,29 @@ def backend(self):
         """Get Hamiltonian's backend."""
         return self.h0.backend
 
-    def hyperopt_step(
+    def least_squares(self, D: np.array):
+        """Least squares cost function."""
+        H = self.h.matrix
+        return -np.real(np.trace(H@D)-0.5*(np.linalg.norm(H)**2+np.linalg.norm(D)**2))
```

Comment:
For a given Hamiltonian we can skip the evaluation of $||H||$ and of $||D||$ which for 10 qubits can already cost some computation time

Reply:
While the norm of H can be removed (just a constant), norm of D should be preserved simply due to the fact that if not the minimum of this function will always be a 0 diagonal. We could maybe optimize the computation of this norm as it is essentially the same as a 2**n vector instead of a 2**n by 2**n matrix and maybe there is a more efficient formula if we consider the magnetic field parametrization